shared/middleware/helpers/assets.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Assets
+#
+class AssetBucket
+
+  ALLOWED_FILE_TYPES = %w(jpg jpeg gif png mp3)
+
+  def initialize
+    params = {region: 'us-east-1'}
+    if CDO.s3_access_key_id && CDO.s3_secret_access_key
+      params[:credentials] = Aws::Credentials.new(CDO.s3_access_key_id, CDO.s3_secret_access_key)
+    end
+    @s3 = Aws::S3::Client.new(params)
```

Comment:
it's possibly slightly more efficient to cache in class variable `@@s3`, since it could reuse connection pools and such rather than recreating them on each new request to the app server. in practice it probably won't make any visible difference either way unless this particular API is a bottleneck at extreme load. if we are already caching other clients in class-variables though, it wouldn't hurt to keep them consistent